mitodl/mitxpro
# View Certificate button and Wagtail handling

#### Pre-Flight checklist


#### What are the relevant tickets?
#940 #875 

#### What's this PR do?
Adds a "View Certificate" button on the dashboard and back-end routing to handle certificate view logic (verifying course and uuid of certificate).

#### How should this be manually tested?
Add a `CourseRunCertificate` object via admin for enrolled courses on a user. Visit the user dashboard and you should see the "View Certificate" button. Clicking on the button should show the certificate as in the screenshots.

N.B.: Ensure that the `CoursePage` for the relevant course has a child `CertificatePage` with required information.

N.B.: Facebook and Twitter share images won't show up until the local xpro deployment is accessible on a public hostname (probably via a tunnel). Note that `ngrok` has trouble with Facebook so staying away from it for testing is advised.

#### Screenshots (if appropriate)

Desktop dashboard
![Screenshot from 2019-09-03 14-58-10](https://user-images.githubusercontent.com/45350418/64163903-8cbc3080-ce5b-11e9-829e-bd198c672318.png)

Mobile dashboard
![screencapture-xpro-odl-local-8053-dashboard-2019-09-03-14_59_01](https://user-images.githubusercontent.com/45350418/64163908-9180e480-ce5b-11e9-95a9-e7374cbad790.png)

Mobile certificate view
![screencapture-sweet-cheetah-18-localtunnel-me-certificates-9a8193e7-2c8d-4d1e-be95-50ece9a4a536-2019-09-11-14_31_32](https://user-images.githubusercontent.com/45350418/64686284-df23cf80-d4a1-11e9-83c1-836a59e3ef68.png)

Desktop certificate view
![screencapture-sweet-cheetah-18-localtunnel-me-certificates-9a8193e7-2c8d-4d1e-be95-50ece9a4a536-2019-09-11-14_31_09](https://user-images.githubusercontent.com/45350418/64686285-df23cf80-d4a1-11e9-9431-ebff32755a11.png)

LinkedIn "Add to Profile" form
![Screenshot from 2019-09-11 14-37-03](https://user-images.githubusercontent.com/45350418/64686305-e77c0a80-d4a1-11e9-9353-f7d7fae2a4cd.png)

Facebook post
![Screenshot from 2019-09-11 14-36-26](https://user-images.githubusercontent.com/45350418/64686306-e814a100-d4a1-11e9-8d40-0084ee203a64.png)

Twitter tweet
![Screenshot from 2019-09-11 14-34-54](https://user-images.githubusercontent.com/45350418/64686307-e814a100-d4a1-11e9-877d-7167baf20ed6.png)

Print dialog
![Screenshot from 2019-09-02 16-20-46](https://user-images.githubusercontent.com/45350418/64111259-aa2ac500-cd9d-11e9-9c13-4dfc2c3fb196.png)
